aeneas/src/ir/SsaNormalizer.v3

```diff
@@ -346,6 +346,306 @@ class SsaRaNormalizer extends SsaRebuilder {
+						return map1(app, result);
+					}
+					x: FloatType => {
+						var width = FloatType.!(ortype).byteSize();
+						for (i < width) {
+							var read = curBlock.opArrayGetElem(an.newType, Int.TYPE, app.facts,
+								newArgs[0], curBlock.opIntAdd(newGraph.intConst(i + offset), pos));
+							var shifted = curBlock.addApply(app.source, Long.TYPE.opShl(),
+								[read, newGraph.intConst(8 * i)]);
+							result = curBlock.addApply(app.source, Long.TYPE.opOr(),
+								[shifted, result]);
+						}
+						var floatViewOp: Operator;
+						if (FloatType.!(ortype).is64) {
+							floatViewOp = V3Op.newFloat64ViewI(result.getType());
+						}
+						else {
+							var intViewOp = V3Op.newIntViewI(result.getType(), Int.TYPE);
+							result = curBlock.addApply(app.source, intViewOp, [result]);
+							floatViewOp = V3Op.newFloat32ViewI(result.getType());
+						}
+						result = curBlock.addApply(app.source, floatViewOp, [result]);
+						return map1(app, result);
+					}
+					_ => ;
+				}
+			}
+			RefStructSetField(offset) => {
+				var newArgs = genRefs(args);
+				var array = newArgs[0], pos = newArgs[1];
+				match(newArgs[2].getType()) {
+					x: IntType => {
+						var width = IntType.!(x).byteSize();
+						//Terminal.put1("%d\n", width);
+						var results = Array<SsaInstr>.new(width);
+						def mask = newGraph.intConst(255);
+						for (i < width) {
+							var shifted = curBlock.addApply(app.source, IntType.!(x).opShr(),
+								[newArgs[2], newGraph.intConst(8 * i)]);
+							var value = curBlock.opIntAnd(shifted, mask);
+							value = curBlock.opIntViewI0(value.getType(), Byte.TYPE, value);
+							var write = curBlock.opArraySetElem(array.getType(), Byte.TYPE, app.facts,
+								newArgs[0], curBlock.opIntAdd(newGraph.intConst(i + offset), pos), value);
+							results[i] = write;
+						}
+						return mapN(app, results);
+					}
+					/* x: EnumType => {
+						var width = EnumType.!(x).byteSize();
+						var results = Array<SsaInstr>.new(width);
+						def mask = newGraph.intConst(255);
+						newArgs[2] = curBlock.opIntViewI0(newArgs[2].getType(), Int.TYPE, newArgs[2]);
+						for (i < width) {
+							var shifted = curBlock.addApply(app.source, Int.TYPE.opShr(),
+								[newArgs[2], newGraph.intConst(8 * i)]);
+							var value = curBlock.opIntAnd(shifted, mask);
+							value = curBlock.opIntViewI0(value.getType(), Byte.TYPE, value);
+							var write = curBlock.opArraySetElem(array.getType(), Byte.TYPE, app.facts,
+								newArgs[0], curBlock.opIntAdd(newGraph.intConst(i + offset), pos), value);
+							results[i] = write;
+						}
+						return mapN(app, results);
+					}*/
+					x: FloatType => {
+						var width = FloatType.!(x).byteSize();
+						var results = Array<SsaInstr>.new(width);
+						def mask = newGraph.longConst(255);
+						newArgs[2] = curBlock.opIntView(newArgs[2].getType(), Long.TYPE, newArgs[2]);
+						for (i < width) {
+							var shifted = curBlock.addApply(app.source, Long.TYPE.opShr(),
```

Comment:
You'll want to use the integer type corresponding to the float type's width. For example `Int.getType(false, x.total_width)`.